### ISSUE TITLE
NaNs in the leaderboard

### ISSUE THREAD

**Issue by schinto** [awslabs/autogluon#646]
In Autogluon version: 0.0.13, I get some NaNs in the leaderboard for the columns `score_val`, `pred_time_val` and `pred_time_val_marginal`. For example see table below.
```
lboard = predictor.leaderboard(silent=True)
lboard.sort_values(by='score_val', ascending=False)
```
**What do these NaNs mean?**

model | score_val | pred_time_val | fit_time | pred_time_val_marginal | fit_time_marginal | stack_level | can_infer | fit_order
-- | -- | -- | -- | -- | -- | -- | -- | --
weighted_ensemble_k0_l2 | 0.921720 | 35.593377 | 337.562476 | 0.002342 | 1.474199 | 2 | True | 22
ExtraTreesClassifierEntr_STACKER_l1 | 0.920474 | 25.178856 | 207.120716 | 3.017217 | 9.877333 | 1 | True | 15
ExtraTreesClassifierGini_STACKER_l1 | 0.917233 | 25.134777 | 207.573372 | 2.973138 | 10.329989 | 1 | True | 14
RandomForestClassifierEntr_STACKER_l1 | 0.916269 | 25.125407 | 208.917588 | 2.963768 | 11.674206 | 1 | True | 13
weighted_ensemble_k0_l1 | 0.913996 | 9.400306 | 152.891460 | 0.002296 | 1.460386 | 1 | True | 11
RandomForestClassifierGini_STACKER_l1 | 0.913840 | 25.094830 | 209.083276 | 2.933191 | 11.839893 | 1 | True | 12
CatboostClassifier_STACKER_l1 | 0.913727 | 24.268665 | 216.692446 | 2.107026 | 19.449063 | 1 | True | 19
LightGBMClassifier_STACKER_l1 | 0.907163 | 24.069759 | 206.304277 | 1.908120 | 9.060894 | 1 | True | 18
LightGBMClassifier_STACKER_l0 | 0.906293 | 1.894278 | 8.656535 | 1.894278 | 8.656535 | 0 | True | 7
LightGBMClassifierCustom_STACKER_l0 | 0.904713 | 1.929491 | 19.792046 | 1.929491 | 19.792046 | 0 | True | 10
CatboostClassifier_STACKER_l0 | 0.904176 | 2.060171 | 24.869649 | 2.060171 | 24.869649 | 0 | True | 8
NeuralNetClassifier_STACKER_l0 | 0.902551 | 2.468340 | 97.097210 | 2.468340 | 97.097210 | 0 | True | 9
RandomForestClassifierEntr_STACKER_l0 | 0.899286 | 2.919619 | 12.814836 | 2.919619 | 12.814836 | 0 | True | 2
RandomForestClassifierGini_STACKER_l0 | 0.898085 | 2.906098 | 11.727416 | 2.906098 | 11.727416 | 0 | True | 1
LightGBMClassifierCustom_STACKER_l1 | 0.897478 | 24.077942 | 217.984738 | 1.916303 | 20.741355 | 1 | True | 21
KNeighborsClassifierDist_STACKER_l1 | 0.894636 | 23.214398 | 198.295784 | 1.052759 | 1.052402 | 1 | True | 17
KNeighborsClassifierUnif_STACKER_l1 | 0.890200 | 23.213356 | 198.296775 | 1.051717 | 1.053392 | 1 | True | 16
KNeighborsClassifierUnif_STACKER_l0 | 0.888856 | 1.045730 | 1.015634 | 1.045730 | 1.015634 | 0 | True | 5
NeuralNetClassifier_STACKER_l1 | 0.880203 | 24.543188 | 284.973389 | 2.381549 | 87.730006 | 1 | True | 20
KNeighborsClassifierDist_STACKER_l0 | 0.876953 | 1.049320 | 1.022043 | 1.049320 | 1.022043 | 0 | True | 6
ExtraTreesClassifierEntr_STACKER_l0 | 0.876567 | 2.935613 | 9.748851 | 2.935613 | 9.748851 | 0 | True | 4
ExtraTreesClassifierGini_STACKER_l0 | 0.875991 | 2.952979 | 10.499163 | 2.952979 | 10.499163 | 0 | True | 3
weighted_ensemble_FULL_k0_l2 | NaN | NaN | 22.515337 | NaN | 0.892156 | 2 | True | 44
weighted_ensemble_FULL_k0_l1 | NaN | NaN | 8.820740 | NaN | 0.686460 | 1 | True | 33
RandomForestClassifierGini_FULL_STACKER_l1 | NaN | NaN | 14.021963 | NaN | 1.247556 | 1 | True | 34
RandomForestClassifierGini_FULL_STACKER_l0 | NaN | NaN | 1.231279 | NaN | 1.231279 | 0 | True | 23
RandomForestClassifierEntr_FULL_STACKER_l1 | NaN | NaN | 14.072553 | NaN | 1.298146 | 1 | True | 35
RandomForestClassifierEntr_FULL_STACKER_l0 | NaN | NaN | 1.236227 | NaN | 1.236227 | 0 | True | 24
NeuralNetClassifier_FULL_STACKER_l1 | NaN | NaN | 17.868088 | NaN | 5.093682 | 1 | True | 42
NeuralNetClassifier_FULL_STACKER_l0 | NaN | NaN | 5.344711 | NaN | 5.344711 | 0 | True | 31
LightGBMClassifier_FULL_STACKER_l1 | NaN | NaN | 13.153849 | NaN | 0.379443 | 1 | True | 40
LightGBMClassifier_FULL_STACKER_l0 | NaN | NaN | 0.434766 | NaN | 0.434766 | 0 | True | 29
LightGBMClassifierCustom_FULL_STACKER_l1 | NaN | NaN | 13.742441 | NaN | 0.968034 | 1 | True | 43
LightGBMClassifierCustom_FULL_STACKER_l0 | NaN | NaN | 0.894646 | NaN | 0.894646 | 0 | True | 32
KNeighborsClassifierUnif_FULL_STACKER_l1 | NaN | NaN | 12.880172 | NaN | 0.105766 | 1 | True | 38
KNeighborsClassifierUnif_FULL_STACKER_l0 | NaN | NaN | 0.125938 | NaN | 0.125938 | 0 | True | 27
KNeighborsClassifierDist_FULL_STACKER_l1 | NaN | NaN | 12.887703 | NaN | 0.113297 | 1 | True | 39
KNeighborsClassifierDist_FULL_STACKER_l0 | NaN | NaN | 0.117510 | NaN | 0.117510 | 0 | True | 28
ExtraTreesClassifierGini_FULL_STACKER_l1 | NaN | NaN | 13.816655 | NaN | 1.042248 | 1 | True | 36
ExtraTreesClassifierGini_FULL_STACKER_l0 | NaN | NaN | 1.020957 | NaN | 1.020957 | 0 | True | 25
ExtraTreesClassifierEntr_FULL_STACKER_l1 | NaN | NaN | 13.802135 | NaN | 1.027728 | 1 | True | 37
ExtraTreesClassifierEntr_FULL_STACKER_l0 | NaN | NaN | 1.034153 | NaN | 1.034153 | 0 | True | 26
CatboostClassifier_FULL_STACKER_l1 | NaN | NaN | 13.718417 | NaN | 0.944010 | 1 | True | 41
CatboostClassifier_FULL_STACKER_l0 | NaN | NaN | 1.334220 | NaN | 1.334220 | 0 | True | 30

**Comment by Innixma** [comment]
This is because you are using `_FULL` models by calling `predictor.refit_full()` or using the `best_quality_with_high_quality_refit` preset. These models are trained on all of the data with no validation set, so they do not have validation scores. Pass in test data to leaderboard to get scores for these models.

In general, these models will have lower accuracy but will have 10x+ reduced inference time and disk usage due to not being bagged ensembles.
